### ISSUE TITLE
Assertion.java:111-115: Avoid the overloading ctor to...

### ISSUE THREAD

**Issue by 0pdd** [llorllale/cactoos-matchers#88]
The puzzle `53-0c767106` from #53 has to be resolved: 

https://github.com/llorllale/cactoos-matchers/blob/84e2fb922eb718274a1d60b57db4ad719f1a6956/src/main/java/org/llorllale/cactoos/matchers/Assertion.java#L111-L115

The puzzle was created by @dgroup on 21-Jan-19. 

  

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0crat** [comment]
Job [#88](https://github.com/llorllale/cactoos-matchers/issues/88) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
The job [#88](https://github.com/llorllale/cactoos-matchers/issues/88) assigned to @longstone[/z](https://www.0crat.com/u/longstone), here is [why](https://www.0crat.com/footprint/C63314D6Z/4ee106ab-c8c6-4dd7-975d-a0f8aad1d9e9); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by longstone** [comment]
@0crat wait
The way the implementation is done creates an impediment. I see the following ways

1. leave it as it is
2. create an AssertException class
3. rebuild the complete mechanism and let exceptions happen during the evaluation and check then if they're expected. this would lead to a lot of new puzzles, I assume.

@llorllale: what is the prefered approach?

**Comment by 0crat** [comment]
> @0crat wait The way the implementation is done creates an impediment. I see the following ways ... ([here](https://github.com/llorllale/cactoos-matchers/issues/88#issuecomment-478243744))

@longstone The impediment for [#88](https://github.com/llorllale/cactoos-matchers/issues/88) was registered successfully by @longstone[/z](https://www.0crat.com/u/longstone)

**Comment by llorllale** [comment]
@longstone none of the above?

I don't have an answer for you, sorry.

**Comment by victornoel** [comment]
@longstone see the discussion here too https://github.com/llorllale/cactoos-matchers/issues/106#issuecomment-482993497

I think things are kind of overcomplicated for nothing.

If it is necessary to keep the `Scalar<T>` argument of `Assertion`, then we should simply have this kind of usage of:
```java
        final AtomicInteger quantity = new AtomicInteger(0);
        new Assertion<Scalar<Text>>(
            "scalar throws the exception",
            () -> () -> {
                quantity.incrementAndGet();
                throw new IllegalStateException("this is a test");
            },
            new Throws<>("this is a test", IllegalStateException.class)
        ).affirm();
```

But I think it would be simply nicer to just have `Assertion` take a `T`.

**Comment by victornoel** [comment]
@longstone for the record, after discussing it in #106, we concluded that the best solution was the one outlined here: https://github.com/llorllale/cactoos-matchers/issues/106#issuecomment-483127231

@llorllale by the way, it seems @longstone stopped working on this…

**Comment by llorllale** [comment]
@0crat refuse

**Comment by llorllale** [comment]
@0crat assign me

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/llorllale/cactoos-matchers/issues/88#issuecomment-484717316))

@llorllale The user @longstone[/z](https://www.0crat.com/u/longstone) resigned from [#88](https://github.com/llorllale/cactoos-matchers/issues/88), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/llorllale/cactoos-matchers/issues/88#issuecomment-484717334))

@llorllale The job [#88](https://github.com/llorllale/cactoos-matchers/issues/88) assigned to @llorllale[/z](https://www.0crat.com/u/llorllale), here is [why](https://www.0crat.com/footprint/C63314D6Z/3e0ee886-c9d0-4f83-b0be-b60c04c3c814); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); we should be aware that @llorllale[/z](https://www.0crat.com/u/llorllale) is on vacation; this ticket may be delayed; there will be no monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by llorllale** [comment]
@0crat wait I submitted PR #109 for this

**Comment by 0crat** [comment]
> @0crat wait I submitted PR #109 for this ([here](https://github.com/llorllale/cactoos-matchers/issues/88#issuecomment-484719357))

@llorllale The impediment for [#88](https://github.com/llorllale/cactoos-matchers/issues/88) was registered successfully by @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0pdd** [comment]
The puzzle `53-0c767106` has disappeared from the source code, that's why I closed this issue.

**Comment by 0pdd** [comment]
@0pdd the puzzle [#110](https://github.com/llorllale/cactoos-matchers/issues/110) is still not solved.

**Comment by 0crat** [comment]
The job [#88](https://github.com/llorllale/cactoos-matchers/issues/88) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +30 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0pdd** [comment]
@0pdd the puzzle [#113](https://github.com/llorllale/cactoos-matchers/issues/113) is still not solved; solved: [#110](https://github.com/llorllale/cactoos-matchers/issues/110).

**Comment by 0pdd** [comment]
@0pdd the puzzle [#116](https://github.com/llorllale/cactoos-matchers/issues/116) is still not solved; solved: [#110](https://github.com/llorllale/cactoos-matchers/issues/110), [#113](https://github.com/llorllale/cactoos-matchers/issues/113).

**Comment by 0pdd** [comment]
@0pdd all 3 puzzles are solved here: [#110](https://github.com/llorllale/cactoos-matchers/issues/110), [#113](https://github.com/llorllale/cactoos-matchers/issues/113), [#116](https://github.com/llorllale/cactoos-matchers/issues/116).
